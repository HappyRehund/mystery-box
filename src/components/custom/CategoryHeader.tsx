import React from "react";
import { Badge } from "@/components/ui/badge";

interface CategoryHeaderProps {
  category: {
    id: string;
    name: string;
    description?: string;
    totalItems?: number;
  };
}

export default function CategoryHeader({ category }: CategoryHeaderProps) {
  return (
    <div className="space-y-2">
      <div className="flex items-center gap-3">
        <h1 className="text-3xl font-bold tracking-tight">{category.name}</h1>
        {category.totalItems && (
          <Badge variant="secondary" className="px-2 py-1">
            {category.totalItems} items
          </Badge>
        )}
      </div>
      {category.description && (
        <p className="text-muted-foreground">{category.description}</p>
      )}
    </div>
  );
}