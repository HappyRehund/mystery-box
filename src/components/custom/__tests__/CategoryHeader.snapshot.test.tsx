import { render } from '@testing-library/react'
import CategoryHeader from '../CategoryHeader'

describe('CategoryHeader Snapshots', () => {
  it('should match snapshot with all props', () => {
    const mockCategory = {
      id: '1',
      name: 'Test Category',
      description: 'Test Description',
      totalItems: 5
    }

    const { container } = render(<CategoryHeader category={mockCategory} />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot without optional props', () => {
    const mockCategory = {
      id: '1',
      name: 'Test Category'
    }

    const { container } = render(<CategoryHeader category={mockCategory} />)
    expect(container).toMatchSnapshot()
  })
})