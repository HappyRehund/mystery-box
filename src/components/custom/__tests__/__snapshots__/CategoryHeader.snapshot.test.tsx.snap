// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryHeader Snapshots should match snapshot with all props 1`] = `
<div>
  <div
    class="space-y-2"
  >
    <div
      class="flex items-center gap-3"
    >
      <h1
        class="text-3xl font-bold tracking-tight"
      >
        Test Category
      </h1>
      <span
        class="inline-flex items-center justify-center rounded-md border text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90 px-2 py-1"
        data-slot="badge"
      >
        5
         items
      </span>
    </div>
    <p
      class="text-muted-foreground"
    >
      Test Description
    </p>
  </div>
</div>
`;

exports[`CategoryHeader Snapshots should match snapshot without optional props 1`] = `
<div>
  <div
    class="space-y-2"
  >
    <div
      class="flex items-center gap-3"
    >
      <h1
        class="text-3xl font-bold tracking-tight"
      >
        Test Category
      </h1>
    </div>
  </div>
</div>
`;
